doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+are currently work in progress. Prepare for changes.
+
+Shoutout:
+* Everyone: Comment! Give us good ideas.
+* libopenage-devs: change/erase this. You are the ones, who (are going to)
+implement this and know what's necessary and what should be where.
+
+```
+Mod():
+	# All patches in the 'patches' member are added on load time. So every mod
+	# needs to inherit from this class and new patches to 'patches'.
+
+	name : text
+	patches : set(NyanPatch) # Add your mod's patches to this member.
+
+Tech():
+	# Every patch that needs to be evaluated after a specific event in the game
+	# (e.g. research) needs to be added to 'updates'.
+
+	name : text              # ingame name
+	description : text       # ingame description
+	updates : set(NyanPatch) # Add your new researches (or other patches) here.
+
+#Proposal: call a `Tech' `Update' in nyan, and define Tech in openage like this:
+Update():
+	updates : set(NyanPatch)
+
+HUDElement():
+	#makes an element on the left of the HUD (building or unit menu)
+	name : text
+	description : text
+	icon: text
+Tech(Update, HUDElement):
+
+
+
+Resource():
+	name : text
+	icon : file
+
+#do these 4 really belong in the engine?
+Gold(Resource)
+Food
+...
+...
+
+
+Building():
+	name : text
+
+Ability(): # WIP
+	# Every unit (WIP: or even buildings, see 'CarryAbility',castles... ), that can do something, has an 'Ability'.
+	# WIP: what members belong here?
+
+HUDAbility(Ability): # WIP
+	# Creates a button on the left of the HUD. For Trebuchet packing and unpacking, unit patrol, formations.
+	name : text
+	description : text
+MoveAbility(Ability):
+	# You want to move? Use this class.
+
+	speed : float
+	instant : bool = False # WIP: how about checking (speed < 0) ⇒ instant before
+	                       # path searching?
+	range : float
+
+CooldownAbility(Ability):
+	# Any executed 'Ability', that inherits from this, can only be executed again
+	# after recharge_time in seconds.
+	recharge_time : float
+
+GatherAbility(Ability): # WIP
+	# Gathering resources requires this Ability
+	# WIP1: do we need a member for the current amount and type of resource here? or is that only inside the engine?
+
+	allowedResource : set(Resource) # Allowed ressources?
+	maxAmount : int # maximum amount  that can be carried
+	speed : float # gathered units per second
+
+	# WIP2: proposal: allow different maximum amounts for different resources:
+
+#	allowedResource : orderedset(Resource)
+#	maxAmount : orderedSet(int) # does <10,5,0,10> work? it is a set after all.
+
+	# WIP3: if WIP2 is not possible: implement an 'Ability' that allows to gather only one resource.
+	# A worker would then need 4 gather abilities added to it's 'abilities' set.
+	# And this would allow to reference
+
+	# WIP4: how do we reference Animations?
+#	animation : file # one picture that contains all frames or, if a directory,
+	                # contains all frames. # WIP5: naming scheme?
```

Comment:
I'd say the animation is a mixin, mainly used in abilities like `Walking(Ability, Animation)`. If that ability becomes active, we can check in the engine if the ability also inherited from `Animation`, then activate and display the data specified there. That way, we can add custom animations for nearly everything we want but do not enforce it and can fall back to the standard walk animation etc.